app/controllers/agents_controller.rb

```diff
@@ -3,30 +3,94 @@ class AgentsController < ApplicationController
   before_action :authorize_and_redirect, :only => [:edit, :update, :create, :new]
 
   def index
-    @agents = LinkedData::Client::Models::Agent.all(include: 'all')
   end
 
   def show
-    # we use :agent_id not :id
-    @agent = LinkedData::Client::Models::Agent.find(params[:agent_id].split('/').last)
-    not_found(t('agents.not_found_agent', id: params[:agent_id])) if @agent.nil?
+        # we use :agent_id not :id
+        @agent = LinkedData::Client::Models::Agent.find(params[:agent_id].split('/').last)
+        not_found(t('agents.not_found_agent', id: params[:agent_id])) if @agent.nil?
+    
+        @agent_id = params[:id] || agent_id(@agent)
+        @name_prefix = params[:name_prefix]
+        @edit_on_modal = params[:edit_on_modal]&.eql?('true')
+        @deletable = params[:deletable]&.eql?('true')
+    end  
+
+  def ajax_agents_list
+    page = params[:page] || 1
+    page_size = params[:pagesize].to_i
+    query = params[:search].presence
+    if query
+      agents = search_agents(query)
+    else
+      options = { page: page, include: 'agentType,name,homepage,acronym,email,identifiers,affiliations,usages' }
+      options[:pagesize] = page_size unless page_size == -1
+  agents = LinkedData::Client::Models::Agent.all(options).first
+    end
 
-    @agent_id = params[:id] || agent_id(@agent)
-    @name_prefix = params[:name_prefix]
-    @edit_on_modal = params[:edit_on_modal]&.eql?('true')
-    @deletable = params[:deletable]&.eql?('true')
+    partial_path = "agents/table"
+    agent_data = agents.collection.map do |agent|
+      # Fallback logic for search result structure
+      name = agent.respond_to?(:name) ? agent.name : agent.name_text
```

Comment:
do `name = agent.name || agent.name_text` and same for bellow lines, but agent.name_text does not exist anymore after your change in the api